lib/engine/game/g_18_india/game.rb

```diff
@@ -963,27 +963,41 @@ def companies_to_payout(ignore: [])
 
         # ----- Route Modificatons for Gauge Change stops (modifed from 1848)
 
-        # Add gauge changes to visited stops, they count as 0 revenue City stops,
+        # Add gauge changes to visited stops, they count as 0 revenue City stops
         def visited_stops(route)
           gauge_changes = border_crossings(route)
           route_stops = route.connection_data.flat_map { |c| [c[:left], c[:right]] }.uniq.compact # super
           return route_stops unless gauge_changes.positive?
 
           LOGGER.debug { "GAME::visited_stops > gauge_changes: #{gauge_changes} route_stops: #{route_stops.inspect}" }
-          add_gauge_changes_to_stops(gauge_changes, route_stops)
+          add_gauge_changes_to_stops(gauge_changes, route_stops, route)
         end
 
-        def add_gauge_changes_to_stops(num, route_stops)
+        def add_gauge_changes_to_stops(num, route_stops, route)
           gauge_changes = Array.new(num) { Engine::Part::City.new('0') }
-          first_stop = route_stops.first
+          gc_tile = gauge_change_tile(route)
           gauge_changes.each do |stop|
-            stop.tile = first_stop.tile
+            stop.tile = gc_tile
             route_stops.insert(1, stop) # add the gauge change after fist element so that it's not the first or last stop
           end
           LOGGER.debug { "GAME::add_gauge_changes_to_stops > route_stops: #{route_stops.inspect}" }
           route_stops
         end
 
+        # return a tile that can be used for a stop, exclude any named hex
+        def gauge_change_tile(route)
+          gc_hexes = @gauge_change_markers.flatten
+          hexes = route.hexes & gc_hexes
+          hexes.each do |hex|
+            return hex.tile unless hex.tile.location_name
+          end
+          # if there isn't an unnamed intersecting hex, use the other GC hexes
+          gc_hexes.each do |hex|
+            return hex.tile unless hex.tile.location_name
+          end
+          raise GameError, 'Tile not found for gauge change'
```

Comment:
This feels like you will have made encountering a bug harder, but not impossible.

Does the gauge change stop's tile actually get used for anything? Could you pass it a dummy Tile object [`Engine::Tile.new(…)`]?

BTW, from seeing this I've realised that you've rolled your own code for checking whether two cities on the same tile are visited in a route. There is already code for this, if you add a 'groups' element to the tile's definition. That would mean you didn't need to worry about tiles with named locations (but it wouldn't have prevented this bug).

Reply:
Actually, using groups to check the route instead of tile names would avoid this bug. Groups are an attribute of the revenue centre, not the tile, so you wouldn't be introducing new stops that had this attribute.

Reply:
I agree with the idea to use a dummy "new" tile. I have figured out what to pass to the method to initialize a new one.

Reply:
OK, I see the 'Groups' definition for revenue centers.  I will test puting those cities into tile groups.

Reply:
Tested using groups, and removed test in `check_other`.

Reply:
Thanks for the suggestions, I think it is better with these adjustments.